include/swift/Serialization/SerializedModuleLoader.h

```diff
@@ -170,10 +169,55 @@ class SerializedModuleLoader : public SerializedModuleLoaderBase {
 public:
   virtual ~SerializedModuleLoader();
 
-  /// Register a memory buffer that contains the serialized
-  /// module for the given access path. This API is intended to be
-  /// used by LLDB to add swiftmodules discovered in the __apple_ast
-  /// section of a Mach-O file to the search path.
+  /// Create a new importer that can load serialized Swift modules
+  /// into the given ASTContext.
+  static std::unique_ptr<SerializedModuleLoader>
+  create(ASTContext &ctx, DependencyTracker *tracker = nullptr,
+         ModuleLoadingMode loadMode = ModuleLoadingMode::PreferSerialized) {
+    return std::unique_ptr<SerializedModuleLoader>{
+      new SerializedModuleLoader(ctx, tracker, loadMode)
+    };
+  }
+};
+
+/// Imports serialized Swift modules from a MemoryBuffer into an ASTContext.
+/// This interface is primarily used by LLDB.
+class MemoryBufferSerializedModuleLoader : public SerializedModuleLoaderBase {
+  llvm::StringMap<std::unique_ptr<llvm::MemoryBuffer>> MemoryBuffers;
+
+  MemoryBufferSerializedModuleLoader(ASTContext &ctx,
+                                     DependencyTracker *tracker,
+                                     ModuleLoadingMode loadMode)
+      : SerializedModuleLoaderBase(ctx, tracker, loadMode) {}
+
+  std::error_code findModuleFilesInDirectory(
+      AccessPathElem ModuleID, StringRef DirPath, StringRef ModuleFilename,
+      StringRef ModuleDocFilename,
+      std::unique_ptr<llvm::MemoryBuffer> *ModuleBuffer,
+      std::unique_ptr<llvm::MemoryBuffer> *ModuleDocBuffer) override {
+    return {};
```

Comment:
This shouldn't return success; it should probably `llvm_unreachable` or something just to be sure.

Reply:
I will return a failure and assert. Unreachable is a bit dangerous in a cross-project API: someone might change the Swift compiler in the future and then cause LLDB to crash in production because the new code path wasn't tested.

Reply:
That's what lldb-moduleimport-test is for, right?

`std::errc::unsupported` is what we've been using elsewhere for "don't even keep searching", I think.